src/Core/Resolvers/Factories/QueryManagerFactory.cs

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Net;
+using Azure.DataApiBuilder.Config.ObjectModel;
+using Azure.DataApiBuilder.Core.Configurations;
+using Azure.DataApiBuilder.Service.Exceptions;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Logging;
+
+namespace Azure.DataApiBuilder.Core.Resolvers.Factories
+{
+    /// <summary>
+    /// QueryManagerFactory. Implements IQueryManagerFactory
+    /// Used to get the appropriate query builder, query executor and exception parser and  based on the database type.
+    /// </summary>
+    public class QueryManagerFactory : IQueryManagerFactory
+    {
+        private readonly IDictionary<DatabaseType, IQueryBuilder> _queryBuilders;
+        private readonly IDictionary<DatabaseType, IQueryExecutor> _queryExecutors;
+        private readonly IDictionary<DatabaseType, DbExceptionParser> _dbExceptionsParsers;
+
+        /// <summary>
+        /// Initiates an instance of QueryManagerFactory
+        /// </summary>
+        /// <param name="runtimeConfigProvider">runtimeconfigprovider.</param>
+        /// <param name="logger">logger.</param>
+        /// <param name="contextAccessor">httpcontextaccessor.</param>
+        public QueryManagerFactory(RuntimeConfigProvider runtimeConfigProvider, ILogger<IQueryExecutor> logger, IHttpContextAccessor contextAccessor)
+        {
+            _queryBuilders = new Dictionary<DatabaseType, IQueryBuilder>();
+            _queryExecutors = new Dictionary<DatabaseType, IQueryExecutor>();
+            _dbExceptionsParsers = new Dictionary<DatabaseType, DbExceptionParser>();
+            foreach (DataSource dataSource in runtimeConfigProvider.GetConfig().ListAllDataSources())
+            {
+                IQueryBuilder? queryBuilder = null;
+                IQueryExecutor? queryExecutor = null;
+                DbExceptionParser? exceptionParser = null;
+
+                if (_queryBuilders.ContainsKey(dataSource.DatabaseType))
```

Comment:
When the size of `_queryBuilders.Count == 3`, it indicates we have created queryBuilders for Ms/Pg/MySql. Can we introduce a break condition when this condition holds true? This would avoid unnecessarily going through all the data sources. (This might be a small optimization considering that the number of data sources should not be too big a number. Obviously, the break condition will need to be adjusted when we support more databases.)

Reply:
I haven't looked deeper into the code, but is there a way to quickly know what all unique database types are present in overall configs? And then instead of looping through all data sources, loop throught those unique db types.

Reply:
This can introduce more complexity for very little performance gain i believe. Currently there are 4 db types but only 3 builders, executors. If you change the number of db's then you need to remember to change the hardcoded 4 or 3 that you set.

Reply:
Yeah, makes sense.